File: Logic_HandsOn/C++/Problems on Digits,String,Array,Pointers etc/program16.cpp
//////////////////////////////////////////////////////////////////////////////////////////////////////
//
//	Program:
//		A program which accept number from user and return the count of even digits.
//
//
/////////////////////////////////////////////////////////////////////////////////////////////////////


#include<iostream>
using namespace std;

class Display
{
	private:
		int iNo;

	public:

		Display(int iNo)
		{
			this->iNo = iNo;
		}	

		int CountEven()
		{	
			int iDigit = 0,iCnt=0;

			if(iNo<0)
			{
				iNo = -iNo;
			}

			while(iNo>0)
			{
				iDigit = iNo%10;
				if(iDigit%2==0)
				{
					iCnt++;
				}
				iNo = iNo/10;
			}

			return iCnt;
		}

};

int main()
{
	int iValue=0,iRet=0;

	cout<<"Enter Number: ";
	cin>>iValue;

	Display *dObj = new Display(iValue);
	iRet = dObj->CountEven();

	cout<<"Even count is "<<iRet<<"\n";
	return 0;
}

/*

	OUTPUT:

	Enter Number: 2395
	Even count is 1

*/

File: Logic_HandsOn/Java/Problems on Array,Pattern etc/Program16.java
//////////////////////////////////////////////////////////////////////////////////////////////////////
//
//	Program:
//		A program which accept number from user and return the count of even digits.
//
//
/////////////////////////////////////////////////////////////////////////////////////////////////////


import java.util.*;

class Solution
{
	public int iValue;

	public Solution(int iValue)
	{
		this.iValue = iValue;
	}

	public int CountEven()
	{
                      	
		int iCnt=0;
		while(iValue!=0)
		{
			int iDigit = iValue%10;
			if(iDigit%2==0)
			{
				iCnt++;
			}
			iValue = iValue/10;
		}        
		return iCnt;
	
	}
}

class Program
{
	public static void main(String[] args)
	{
		Scanner sObj = new Scanner(System.in);

		int iRet=0;
		System.out.println("Enter number: ");
		int iNo = sObj.nextInt();

		Solution obj = new Solution(iNo);
		iRet = obj.CountEven();
	
		System.out.println("Even count is: "+iRet);
	}
}

/*
OUTPUT:

Enter number:
8462
Even count is: 4

*/
